drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/DecisionTableAnalyzerProvider.java

```diff
@@ -0,0 +1,115 @@
+import org.drools.workbench.screens.guided.dtable.client.widget.analysis.cache.RuleInspectorCache;
+import org.drools.workbench.screens.guided.dtable.client.widget.analysis.cache.UpdateManager;
+import org.drools.workbench.screens.guided.dtable.client.widget.analysis.checks.base.CheckRunner;
+import org.drools.workbench.screens.guided.dtable.client.widget.analysis.index.Index;
+import org.drools.workbench.screens.guided.dtable.client.widget.analysis.index.IndexBuilder;
+import org.drools.workbench.screens.guided.dtable.client.widget.analysis.panel.AnalysisReportScreen;
+import org.drools.workbench.screens.guided.dtable.client.widget.table.utilities.ColumnUtilities;
+import org.drools.workbench.screens.guided.dtable.service.GuidedDecisionTableEditorService;
+import org.kie.workbench.common.services.shared.preferences.ApplicationPreferences;
+import org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracle;
+import org.uberfire.mvp.PlaceRequest;
+
+@Dependent
+public class DecisionTableAnalyzerProvider {
+
+    private final AnalysisReportScreen analysisReportScreen;
+
+    @Inject
+    public DecisionTableAnalyzerProvider( final AnalysisReportScreen analysisReportScreen ) {
+        this.analysisReportScreen = analysisReportScreen;
+    }
+
+    public AnalyzerController newAnalyzer( final PlaceRequest placeRequest,
+                                           final AsyncPackageDataModelOracle oracle,
+                                           final GuidedDecisionTable52 model,
+                                           final EventBus eventBus ) {
+        if ( isAnalysisEnabled() ) {
+            return makeTheRealThing( placeRequest,
+                                     oracle,
+                                     model,
+                                     eventBus );
+        } else {
+            return makePlaceHolder();
+        }
+    }
+
+    private boolean isAnalysisEnabled() {
+        if ( ApplicationPreferences.getStringPref( GuidedDecisionTableEditorService.DTABLE_VERIFICATION_ENABLED ) != null ) {
+            return ApplicationPreferences.getBooleanPref( GuidedDecisionTableEditorService.DTABLE_VERIFICATION_ENABLED );
+        } else {
+            return true;
+        }
+    }
+
+    private AnalyzerController makePlaceHolder() {
+        return new AnalyzerController() {
+            @Override
+            public void initialiseAnalysis() {
+
+            }
+
+            @Override
+            public void terminateAnalysis() {
+
+            }
+        };
+    }
+
+    private AnalyzerControllerImpl makeTheRealThing( final PlaceRequest placeRequest,
+                                                     final AsyncPackageDataModelOracle oracle,
+                                                     final GuidedDecisionTable52 model,
+                                                     final EventBus eventBus ) {
+        final CheckRunner checkRunner = new CheckRunner();
```

Comment:
This is a lot of wiring.. I'm surprised you didn't write a "builder"! (not that I'm objecting to the PR on this basis)